androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -232,9 +233,32 @@ class AgendaDetailsViewModel(
 
     private fun saveUpdate() {
         viewModelScope.launch {
-            agendaRepository.updateAgenda(screenStateFlow.value.agendaItem).onSuccess {
-                _screenStateFlow.update {
-                    it.copy(isEdit = false)
+            val agendaItem = screenStateFlow.value.agendaItem
+
+            when (agendaItem) {
+                is Task -> agendaRepository.updateTask(agendaItem)
+                is Reminder -> agendaRepository.updateReminder(agendaItem)
+                is Event -> {
+                    val eventIsGoing = screenStateFlow.value.eventIsGoing
+                    agendaRepository.updateEvent(
+                        body =
+                            UpdateEventBody(
+                                event = agendaItem,
+                                deletedPhotoKeys = deletedPhotoKeys,
+                                isGoing = eventIsGoing,
+                                photos = emptyList(),
+                            ),
+                    )
```

Comment:
Per my previous comment, these body models belong in your data layer as they model the data required by API endpoints. The domain layer should not be aware of them as they are implementation details.